domains/service/src/network.rs

```diff
@@ -0,0 +1,209 @@
+use sp_api::__private::BlockT;
+use sp_blockchain::{HeaderBackend, HeaderMetadata};
+use sp_consensus::block_validation::{Chain, DefaultBlockAnnounceValidator};
+use sp_domains::DomainsApi;
+use sp_runtime::traits::{BlockIdTo, Header};
+use std::sync::Arc;
+
+pub mod execution_receipt_protocol;
+pub mod receipt_receiver;
+
+/// Build the network service, the network status sinks and an RPC sender.
+#[allow(clippy::type_complexity)]
+pub fn build_network<Block, Net, TxPool, IQ, Client, CBlock, CClient, DomainHeader>(
+    params: BuildNetworkParams<Block, Net, TxPool, IQ, Client>,
+    consensus_client: Arc<CClient>,
+) -> Result<
+    (
+        Arc<dyn sc_network::service::traits::NetworkService>,
+        TracingUnboundedSender<sc_rpc::system::Request<Block>>,
+        sc_network_transactions::TransactionsHandlerController<<Block as BlockT>::Hash>,
+        NetworkStarter,
+        Arc<SyncingService<Block>>,
+        NetworkServiceHandle,
+        Arc<dyn BlockDownloader<Block>>,
+    ),
+    Error,
+>
+where
+    Block: BlockT<Header = DomainHeader>,
+    CBlock: BlockT,
+    Client: ProvideRuntimeApi<Block>
+        + HeaderMetadata<Block, Error = sp_blockchain::Error>
+        + Chain<Block>
+        + BlockBackend<Block>
+        + BlockIdTo<Block, Error = sp_blockchain::Error>
+        + ProofProvider<Block>
+        + HeaderBackend<Block>
+        + BlockchainEvents<Block>
+        + AuxStore
+        + 'static,
+    TxPool: TransactionPool<Block = Block, Hash = <Block as BlockT>::Hash> + 'static,
+    IQ: ImportQueue<Block> + 'static,
+    Net: NetworkBackend<Block, <Block as BlockT>::Hash>,
+    CClient: ProvideRuntimeApi<CBlock>
+        + BlockBackend<CBlock>
+        + ProofProvider<CBlock>
+        + HeaderBackend<CBlock>
+        + Send
+        + Sync
+        + 'static,
+    CClient::Api: DomainsApi<CBlock, DomainHeader>,
+    DomainHeader: Header,
+{
+    let BuildNetworkParams {
+        config,
+        mut net_config,
+        client,
+        transaction_pool,
+        spawn_handle,
+        import_queue,
+        block_announce_validator_builder,
+        warp_sync_config,
+        block_relay,
+        metrics,
+    } = params;
+    let fork_id = config.chain_spec.fork_id();
+
+    let block_announce_validator = if let Some(f) = block_announce_validator_builder {
+        f(client.clone())
+    } else {
+        Box::new(DefaultBlockAnnounceValidator)
+    };
+
+    // "Last confirmed domain block execution receipt" request handler
+    {
+        let num_peer_hint = net_config.network_config.default_peers_set_num_full as usize
+            + net_config
+                .network_config
+                .default_peers_set
+                .reserved_nodes
+                .len();
+
+        let (handler, protocol_config) = LastDomainBlockERRequestHandler::new::<Net>(
+            fork_id,
+            consensus_client,
+            client.clone(),
+            num_peer_hint,
+        );
+        spawn_handle.spawn(
```

Comment:
Ok, so I think I’m missing some context here.

What’s the impact on Subspace if each of these tasks fails?
- last-domain-execution-receipt-request-handler
- block-request-handler
- syncing

Let’s assume a failure on most or all of the nodes on the network, because that could easily happen in the logs and we wouldn’t know about it. (Because the node won’t go down, and most people don’t read logs.)